interface/controller/app_controller.go

```diff
@@ -0,0 +1,107 @@
+package controller
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/lscantillowl/academy-go-q42021/domain/model"
+)
+
+// respondWithJSON write json response format to the response writer
+func respondWithJSON(w http.ResponseWriter, code int, payload interface{}) {
+	response, err := json.Marshal(payload)
+	if err != nil {
+		log.Fatalf("Can not convert payload to JSON - %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	w.Write(response)
+}
+
+// Home handler to test the api
+func HomeHandler(w http.ResponseWriter, r *http.Request) {
+	resp := make(map[string]string)
+	resp["message"] = "Welcome to the Q4 GO Bootcamp API"
+	respondWithJSON(w, http.StatusOK, resp)
+}
+
+// Function to get characters from csv file and return it as json
+func GetCharacters(w http.ResponseWriter, r *http.Request) {
+
+	file, err := os.Open("characters.csv")
```

Comment:
I will suggest two things:

1. Make this match to the save method, as the final delivery was though doing the following:

- Call the external API to store in this case a pokemon
- The other endpoint should be able to retrieve this data, otherwise the project would not make that much sense of having to independent sources.

2. The final delivery would consist on making the read using concurrency, and instead of filling your CSV manually, it would be helpful doing it by calling your second endpoint.